app/main.py

```diff
@@ -1 +1,44 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name:
+                 str, health:
+                 int = 100, hidden:
+                 bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}, " \
+               f"Health: {self.health}, " \
+               f"Hidden: {self.hidden}}}"
+
+    @classmethod
+    def del_dead_animal(cls) -> None:
+        cls.alive = [animal for animal in Animal.alive if animal.health > 1]
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        if self.hidden is False:
+            self.hidden = True
+        else:
+            self.hidden = False
```

Comment:
You can simplify this code
```suggestion
        self.hidden = not self.hidden
```